### PR TITLE
Github actions to bump version and execute tests added

### PR DESCRIPTION
**Pull request by lucvanderzandt**
Changes in this pull request:

- Extended pull request #7: moved API keys to environment variables and build in a check to see if they're present before executing a test, otherwise skipping them.
**!! @ Maintainers (@marcelcorso @Multimo a.o.): please add an API test key, API live key and MSISDN to the repo secrets to execute the tests. !!**

- ~~Added a Github Action to automatically bump the version (minor) when pushing a change to master. See the [docs](https://github.com/anothrNick/github-tag-action) of this action for more info.
**!! Please increase the version number in the pubspec as well and specify your changes in the CHANGELOG.md. !!**~~
Changed by commit #86b8166

### CODE REVIEW

**Comment by lucvanderzandt** [comment]
I also leveled the feature branch with master, as you can see. So no conflicts with master if I'm correct.

**Comment by lucvanderzandt** [comment]
Just FYI: I didn't yet fully master getting the keys from Github Secrets, so a few tests were skipped in the original PR. I extended the PR and now it's working. For verification, see the latest Dart CI builds at [lucvanderzandt/dart-rest-api/actions](https://github.com/lucvanderzandt/dart-rest-api/actions), where all tests are passing (none skipped).

**Comment by marcelcorso** [comment]
thanks @lucvanderzandt 🎉

File: README.md [hunk 1]
```diff
@@ -37,27 +42,31 @@ void main() {
...
-    "test": "YOUR_API_TEST_KEY",
-    "live": "YOUR_API_LIVE_KEY",
-    "msisdn": "YOUR_MSISDN"
-}
+Before running the tests, make sure you have the following environment variables in your system:
+
+```
+"API_TEST_KEY": "YOUR_API_TEST_KEY",
+"API_LIVE_KEY": "YOUR_API_LIVE_KEY",
+"MSISDN": "YOUR_MSISDN"
```
Comment (Multimo) [inline README.md]: What do you think about having an example for the msisdn? might not be super obvious for everyone and numbers can come in a few different formats.

File: test/callflows/callflows_service_test.dart [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-    Map credentials;
     CallflowsService callflowsService;
     Callflow callflow;
     String id;
 
     setUp(() {
-      credentials =
-          json.decode(File('test_resources/keys.json').readAsStringSync());
-      callflowsService = ApiCallflowsService(credentials['live']);
+      callflowsService = ApiCallflowsService(credentials.API_LIVE_KEY);
```
Comment (Multimo) [inline test/callflows/callflows_service_test.dart]: Do you know if its possible to use the TEST keys here instead?

**Review by Multimo — APPROVED** [review]
Excellent work @lucvanderzandt 👍 

I have added the keys to the project so hopefully it will work when we merge it. Please let me know when you think its ready to be merged.

**Comment by Multimo** [comment]
Something for another pr but I wonder if its possible to automate this somehow:
!! Please increase the version number in the pubspec as well and specify your changes in the CHANGELOG.md. !!

File: README.md (hunk 1 above)
Comment (lucvanderzandt) [reply README.md]: Sure! That could be incorporated in the readme. Maybe a table providing some an explanation and some examples for these values.

File: test/callflows/callflows_service_test.dart (hunk 2 above)
Comment (lucvanderzandt) [reply test/callflows/callflows_service_test.dart]: I don't know by heart, but in my experience what you can do with the test key is very limited. It mainly has the purpose of verifying that the connection has been established.

Comment (lucvanderzandt) [reply test/callflows/callflows_service_test.dart]: Nevermind! I was wrong, new commits incoming where the callflow service uses the test key

**Comment by lucvanderzandt** [comment]
> Something for another pr but I wonder if its possible to automate this somehow:
> !! Please increase the version number in the pubspec as well and specify your changes in the CHANGELOG.md. !!

I just pushed a few commits with a GitHub action to tag all previous commits with the version number specified in the pubspec and update the readme with the commit messages. It's ready to merge into master now.

**Pull request merged**